test/e2e/api_bulk.go

```diff
@@ -3,188 +3,208 @@
 package test_suite
 
 import (
+	"math/big"
+	"time"
+
 	"github.com/formancehq/go-libs/v2/logging"
 	. "github.com/formancehq/go-libs/v2/testing/api"
 	"github.com/formancehq/go-libs/v2/testing/platform/pgtesting"
 	. "github.com/formancehq/ledger/pkg/testserver"
 	"github.com/formancehq/stack/ledger/client/models/components"
 	"github.com/formancehq/stack/ledger/client/models/operations"
-	"math/big"
-	"time"
 
 	"github.com/formancehq/go-libs/v2/metadata"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 )
 
 var _ = Context("Ledger engine tests", func() {
-	var (
-		db          = pgtesting.UsePostgresDatabase(pgServer)
-		ctx         = logging.TestingContext()
-		bulkMaxSize = 5
-	)
 
-	testServer := NewTestServer(func() Configuration {
-		return Configuration{
-			PostgresConfiguration: db.GetValue().ConnectionOptions(),
-			Output:                GinkgoWriter,
-			Debug:                 debug,
-			NatsURL:               natsServer.GetValue().ClientURL(),
-			BulkMaxSize:           bulkMaxSize,
-		}
-	})
-	BeforeEach(func() {
-		err := CreateLedger(ctx, testServer.GetValue(), operations.V2CreateLedgerRequest{
-			Ledger: "default",
-		})
-		Expect(err).To(BeNil())
-	})
-	When("creating a bulk on a ledger", func() {
-		var (
-			now   = time.Now().Round(time.Microsecond).UTC()
-			items []components.V2BulkElement
-			err   error
-		)
-		BeforeEach(func() {
-			items = []components.V2BulkElement{
-				components.CreateV2BulkElementCreateTransaction(components.V2BulkElementCreateTransaction{
-					Data: &components.V2PostTransaction{
-						Metadata: map[string]string{},
+	for _, data := range []struct {
+		description      string
+		numscriptRewrite bool
+	}{
+		{"default", false},
+		{"numscript rewrite", true},
+	} {
+
+		Context(data.description, func() {
+			var (
+				db          = pgtesting.UsePostgresDatabase(pgServer)
+				ctx         = logging.TestingContext()
+				bulkMaxSize = 5
+			)
+
+			testServer := NewTestServer(func() Configuration {
+				return Configuration{
+					PostgresConfiguration:        db.GetValue().ConnectionOptions(),
+					Output:                       GinkgoWriter,
+					Debug:                        debug,
+					NatsURL:                      natsServer.GetValue().ClientURL(),
+					BulkMaxSize:                  bulkMaxSize,
+					ExperimentalNumscriptRewrite: data.numscriptRewrite,
+				}
+			})
+			BeforeEach(func() {
+				err := CreateLedger(ctx, testServer.GetValue(), operations.V2CreateLedgerRequest{
+					Ledger: "default",
+				})
+				Expect(err).To(BeNil())
+			})
+			When("creating a bulk on a ledger", func() {
+				var (
+					now   = time.Now().Round(time.Microsecond).UTC()
```

Comment:
Not a big deal but there is wrapper around `time` package in go-libs, which already round dates at microseconds (postgres precision)

Reply:
and set UTC too